porcupine/plugins/markdown.py

```diff
@@ -0,0 +1,56 @@
+"""If configuration says so, insert spaces when the tab key is pressed."""
+
+from __future__ import annotations
+
+import logging
+import re
+import tkinter
+from functools import partial
+
+from porcupine import get_tab_manager, tabs, textutils, utils
+
+log = logging.getLogger(__name__)
+
+
+setup_before = ["tabs2spaces"]
+
+
+def _is_list_item(line: str) -> bool:
+    """Detect if the line that is passed is a markdown list item
+
+    According to:
+    - https://spec.commonmark.org/0.30/#lists
+    - https://pandoc.org/MANUAL.html#lists
+    """
+    assert len(line.splitlines()) == 1
+    pattern = r"(^\s*\d{1,9}[.)]|^\s*[-+*]|^\s*#\)|^\s*#\.) .*"
+    regex = re.compile(pattern)
+    match = regex.search(line)
+    return bool(match)
+
+
+def on_tab_key(
+    tab: tabs.FileTab, event: tkinter.Event[textutils.MainText], shift_pressed: bool
+) -> str | None:
+    """Indenting and dedenting list items"""
+    if tab.settings.get("filetype_name", str) == "Markdown":
```

Comment:
Maybe also check what the character just before the cursor is. If it is not a space character, we probably don't want to indent, so that you can autocomplete annoyingly long words even if you are writing that word into a list item. Something like `tab.textwidget.get("insert - 1 char", "insert") == " "`

Reply:
Makes sense, will do that

Reply:
I would think that dedenting would make sense even if it's isn't a space char prior, because `Shift-Tab` doesn't trigger the autocomplete?

Reply:
Fixed with a1479f6a9e0b3631fa10b946b5a12627a7fb9e71